src/main/java/com/google/cloud/tools/ide/login/GoogleLoginState.java

```diff
@@ -315,23 +324,35 @@ private void notifyLoginStatusChange() {
   @VisibleForTesting
   UserInfo queryUserInfo(Credential credential)
       throws IOException, EmailAddressNotReturnedException {
-    HttpRequestInitializer timeoutSetter = new HttpRequestInitializer() {
-      @Override
-      public void initialize(HttpRequest httpRequest) throws IOException {
-        httpRequest.setConnectTimeout(USER_INFO_QUERY_HTTP_CONNECTION_TIMEOUT);
-        httpRequest.setReadTimeout(USER_INFO_QUERY_HTTP_READ_TIMEOUT);
-      }
-    };
 
-    UserInfo userInfo = userInfoService.buildAndExecuteRequest(
-        transport, jsonFactory, credential, timeoutSetter);
+    Oauth2 oAuth2 = buildOAuth2(credential);
+    UserInfo userInfo = oAuth2Wrapper.executeOAuth2(oAuth2);
     if (userInfo == null || Strings.isNullOrEmpty(userInfo.getEmail())) {
       throw new EmailAddressNotReturnedException("Server failed to return email address");
     }
     return userInfo;
   }
 
   /**
+   * Builds an {@link Oauth2} instance carrying auth info from {@code credential}. Additionally sets
+   * connection and read timeouts.
+   */
+  @VisibleForTesting
+  Oauth2 buildOAuth2(final Credential credential) {
```

Comment:
Yes, though it still bothers me that the capitalization is inconsistent. If we go with OAuth we need to make very sure we don't introduce both OAuthUser and OauthUser or anything like that.